packages/app/src/messages/components/messages/messages.component.tsx

```diff
@@ -172,50 +184,54 @@ const Messages: FunctionComponent<MessagesProps> = ({
   }, [messageLayoutNotifications])
 
   useEffect(() => {
-    const hideInfoPopupsTimeout = setTimeout(() => {
-      messagePopupNotifications
-        .filter((item) => item.method === NotificationMethod.Popup)
-        .forEach((item) => {
-          removeNotification(item.id)
-        })
+    messagePopupNotifications.length > 0 &&
+      removePopupNotification(messagePopupNotifications)
+    threadPopupNotifications.length > 0 &&
+      removePopupNotification(threadPopupNotifications)
+    // AUTO DISABLED - fix me if you like :)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  useEffect(() => {
+    if (
+      state === State.Loaded &&
+      messagePopupNotifications.length > 0 &&
+      states.messageDeleting
+    ) {
+      updateFieldState("messageDeleting", false)
+      updateFieldState("messageDeletingConfirmation", false)
```

Comment:
At least for me, it is not intuitive that deleting state and search state depends on the state/change of `messagePopupNotifications` value. The name `messagePopupNotifications` indicates that this is just a general notification, not even strictly bounded with a delete operation (which is fine).

Maybe we could move the above `if` to the separate `useEffect` that would depend on something else than `messagePopupNotifications`? Or feel free to add any other solution.

(similar case to `threadPopupNotifications`).

Reply:
This events depends on `messagePopupNotifications` as a part of deleting modals flow, notification is kind of one of them.  Checking if `messagesState` is `searchResult` is for refresh searched results after deleting message.

Reply:
My point is - `notification` is something that should be just displayed and hidden after some time. IMHO the fact that checking if e.g. message is already deleted and we if should hide confirmation popup depends on the notification, is a little bit unexpected.

Not forcing though, now the names (`messageDeleteNotifications`, `threadDeleteNotifications`) indicate its bounding with deletion feature + maybe it is the most reasonable way to achieve correct behaviour.